File: backend/models/rental_property.go
package models

type RentalProperty struct {
	ID          string   `json:"id" firestore:"id"`
	Name        string   `json:"name" firestore:"name"`
	Description string   `json:"description" firestore:"description"`
	Address     string   `json:"address" firestore:"address"`
	Price       float64  `json:"price" firestore:"price"`
	Bedrooms    int      `json:"bedrooms" firestore:"bedrooms"`
	Bathrooms   int      `json:"bathrooms" firestore:"bathrooms"`
	Area        float64  `json:"area" firestore:"area"`
	ImageURLs   []string `json:"imageUrls" firestore:"imageUrls"`
}

File: frontend/src/types/RentalProperty.ts
export interface RentalProperty {
  id: string;
  name: string;
  description: string;
  address: string;
  price: number;
  imageUrls: string[];
  amenities: string[];
  maxGuests: number;
  bedrooms: number;
  bathrooms: number;
  location: string;
  propertyType: string;
  rating: number;
  reviewCount: number;
  availableDates: {
    start: string;
    end: string;
  };
  host: {
    name: string;
    rating: number;
    responseRate: number;
  };
  area: number;
}
